### PR TITLE
DOC: Added examples to the IndexOpsmixin.value_counts() docstring

### PR DESCRIPTION
**Pull request by marcosrullan**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [ ] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```

###############################################################################
#################### Docstring (pandas.Index.value_counts)  ####################
################################################################################

Return object containing counts of unique values.

The resulting object will be in descending order so that the
first element is the most frequently-occurring element.
Excludes NA values by default.

Parameters
----------
normalize : boolean, default False
    If True then the object returned will contain the relative
    frequencies of the unique values.
sort : boolean, default True
    Sort by values.
ascending : boolean, default False
    Sort in ascending order.
bins : integer, optional
    Rather than count values, group them into half-open bins,
    a convenience for pd.cut, only works with numeric data.
dropna : boolean, default True
    Don't include counts of NaN.

Returns
-------
counts : Series

Examples
--------

>>> index = pd.Index([3, 1, 2, 3, 4, np.nan])
>>> index.value_counts()
3.0    2
4.0    1
2.0    1
1.0    1
dtype: int64

With `normalize` set to `True`, returns the relative frequency by
dividing all values by the sum of values.

>>> s = pd.Series([3, 1, 2, 3, 4, np.nan])
>>> s.value_counts(normalize=True)
3.0    0.4
4.0    0.2
2.0    0.2
1.0    0.2
dtype: float64

**bins**

Bins can be useful for going from a continuous variable to a
categorical variable; instead of counting unique
apparitions of values, divide the index in the specified
number of half-open bins.

>>> s.value_counts(bins=3)
(2.0, 3.0]      2
(0.996, 2.0]    2
(3.0, 4.0]      1
dtype: int64

**dropna**

With `dropna` set to `False` we can also see NaN index values.

>>> s.value_counts(dropna=False)
3.0    2
NaN     1
4.0    1
2.0    1
1.0    1
dtype: int64

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
	See Also section not found

```
The validation is not passing because we lack an see also section, but we think that is not relevant for this method.

### CODE REVIEW

File: pandas/core/base.py [hunk 1]
```diff
@@ -943,18 +943,65 @@ def value_counts(self, normalize=False, sort=True, ascending=False,
...
         dropna : boolean, default True
             Don't include counts of NaN.
 
         Returns
         -------
         counts : Series
+
+        Examples
+        --------
+
```
Comment (cemsbr) [inline pandas/core/base.py]: I've noticed there's no blank line after section headers in https://python-sprints.github.io/pandas/guide/pandas_docstring.html

**Review by cemsbr — CHANGES_REQUESTED** [review]

**Comment by caperutxa** [comment]
Pa que tocas

File: pandas/core/base.py (hunk 1 above)
Comment (marcosrullan) [reply pandas/core/base.py]: Thanks, I have removed the unnecessary new line!

**Review by cemsbr — APPROVED** [review]

**Comment by jreback** [comment]
thanks @marcosrullan and @mroeschke for the fixups!

**Pull request merged**